src/rtf-plugins/fixture-managers/yarpserver/yarpserverfixture.cpp

```diff
@@ -49,7 +49,13 @@ class YarpNameServer : public FixtureManager
     {
 
         yServer.configure(argc, argv);
-        return yServer.start();
+        yServer.start();
+        double time = yarp::os::Time::now();
+        while(!net.checkNetwork() || (yarp::os::Time::now() - time) < YSERVFIXT_TIMEOUT)
+        {
+            yarp::os::Time::delay(0.1);
```

Comment:
Here as well: 
~~~
yarp::os::SystemClock::delaySystem(0.1)
~~~
Furthermore, I would suggest to choose a smaller waiting time, such as `0.001`